rl4co/envs/routing/cvrptw.py

```diff
@@ -0,0 +1,162 @@
+from typing import Optional
+import torch
+from tensordict.tensordict import TensorDict
+from torchrl.data import (
+    BoundedTensorSpec,
+    CompositeSpec,
+    UnboundedContinuousTensorSpec,
+    UnboundedDiscreteTensorSpec,
+)
+from zmq import device
+
+from rl4co.envs.routing.cvrp import CVRPEnv, CAPACITIES
+from rl4co.utils.ops import gather_by_index, get_distance
+
+
+class CVRPTWEnv(CVRPEnv):
+    """
+    An implementation of the Capacitated Vehicle Routing Problem (CVRP) with Time Windows (CVRPTW) environment.
+    Inherits from the CVRPEnv class.
+    """
+
+    name = "cvrptw"
+
+    def __init__(
+        self,
+        min_time: float = 0.0,
+        max_time: float = 100.0,
```

Comment:
Is this a usual value for the CVRPTW problem environment?
Since we are considering 1 unit of time = 1 unit of distance, `100` may be too large ?

Reply:
No idea, I didn’t think too much about a value that would make sense, but you’re right, it’s probably way too large. If we consider time=distance, maybe a value between 8-16 hours could make sense to simulate a full working day (with several shifts in the case of max_time>8).

Reply:
I see, let's set this afterward when we make the data generation part then 👍🏼 
One idea is the one used [here](https://github.com/jokofa/JAMPR/blob/master/data_generator.py) , i.e. making some scaling. I think the best way would be to take some instances from challenges (i.e. [this](https://wouterkool.github.io/pdf/paper-kool-hgs-vrptw.pdf)) such as the SOLOMON ones and test out whether they can easily fit